src/pathfinder/yapf/yapf_ship.cpp

```diff
@@ -388,6 +388,13 @@ class CYapfCostShipT {
 		uint8_t speed_frac = (GetEffectiveWaterClass(n.GetTile()) == WATER_CLASS_SEA) ? svi->ocean_speed_frac : svi->canal_speed_frac;
 		if (speed_frac > 0) c += YAPF_TILE_LENGTH * (1 + tf->tiles_skipped) * speed_frac / (256 - speed_frac);
 
+		/* Lock penalty. */
+		if (IsTileType(n.GetTile(), MP_WATER) && IsLock(n.GetTile()) && GetLockPart(n.GetTile()) == LOCK_PART_MIDDLE) {
+			const uint canal_speed = svi->ApplyWaterClassSpeedFrac(svi->max_speed, false);
+			/* Cost is proportional to the vehicle's speed as the vehicle stops in the lock. */
+			c += (4 * YAPF_TILE_LENGTH * canal_speed) / TILE_HEIGHT / 8;
```

Comment:
Does `TILE_HEIGHT` actually affect lock speed? (ships don't gain/lose elevation at different rates, do they?)

Reply:
Upon retrospect the division by `TILE_HEIGHT` and frame counts is just plain wrong. If we were to double `TILE_HEIGHT` or frame counts, the cost should get higher... not lower. 

So the change to the cost calculation needs to be reverted ;(

Reply:
Yeah good point, should have thought of that myself. I changed it again but kept TILE_HEIGHT in, this time on the enumerator side.